portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that gets data from a csv and parses it as to use it in a chart.*/
+@WebServlet("/chart-data")
+public class ChartServlet extends HttpServlet {
+  private LinkedHashMap<String, Integer> memeSearchTrend = new LinkedHashMap<>();
+
+  /** Parses csv data a single time instead of on every request.*/
+  @Override
+  public void init() {
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream(
+        "/WEB-INF/meme-trend.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      String date = cells[0];
+      Integer interest = Integer.valueOf(cells[1]);
+
```

Comment:
nit: Not sure you need to put these extra line spaces here!

Reply:
Done!